src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -67,6 +71,26 @@ public static void AddTagsFromDbCommand(this Span span, IDbCommand command)
             }
         }
 
+        internal static void SetHeaderTags(this Span span, IHeadersCollection headers, IDictionary<string, string> headerTags, string defaultTagPrefix)
+        {
+            if (headerTags is not null && !headerTags.IsEmpty())
+            {
+                try
+                {
+                    // extract propagation details from http headers
+                    var tagsFromHeaders = SpanContextPropagator.Instance.ExtractHeaderTags(headers, headerTags, defaultTagPrefix);
+                    foreach (KeyValuePair<string, string> kvp in tagsFromHeaders)
+                    {
+                        span.SetTag(kvp.Key, kvp.Value);
```

Comment:
Possible optimization for later: add `SetTags(tagsFromHeaders)` which uses `Dictionary<TKey,TValue>.EnsureCapacity(Int32)` (added in .NET Core 2.1) to increase the dictionary size only once.